memote/support/matrix.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2017 Novo Nordisk Foundation Center for Biosustainability,
+# Technical University of Denmark.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Supporting functions checking the matrix condition of the model object."""
+
+from __future__ import absolute_import
+
+import logging
+import numpy as np
+import memote.support.consistency_helpers as con_helpers
+
+LOGGER = logging.getLogger(__name__)
+
+
+def absolute_extreme_coefficient_ratio(model):
+    """Return the absolute maximum & absolute non-zero minimum coefficients."""
+    # S-Matrix with absolute values:
+    s_matrix, _, _ = con_helpers.stoichiometry_matrix(
+        model.metabolites, model.reactions
+    )
+    abs_matrix, _, _ = abs(s_matrix)
+
+    absolute_max_coef = abs_matrix.max()
+    absolute_non_zero_min_coef = np.min(abs_matrix[np.nonzero(abs_matrix)])
+
+    return (absolute_max_coef, absolute_non_zero_min_coef)
+
+
+def number_independent_conservation_relations(model):
```

Comment:
Given by the name I'd expect this function to return an integer of the number of independent conservation relations and not the left nullspace matrix itself.

It generally makes more sense to put as much of the logic into these helper functions and only pass the final result to the tests. This is because we currently don't unit test the tests themselves.

Also see my comment above in the corresponding test (`test_number_independent_conservation_relations`).

Reply:
The thing is that currently `data` can only be displayed on the reports if it is one of the following (quoted from `annotate` docstring in `utils.py`):

 - 'number' : 'data' is a single number which can be an integer or
          float and should be represented as such.
 - 'count' : 'data' is a list, set or tuple. Choosing 'count' will
          display the length of that list e.g. number of metabolites without formula.
 - 'percent' : Instead of 'data' the content of 'metric' ought to be
          displayed e.g. percentage of metabolites without charge.
          'metric' is expected to be a floating point number.
- 'raw' : 'data' is ought to be displayed "as is" without formatting.
          This option is appropriate for single strings or a boolean output.